### ISSUE TITLE
BooleanPropertyNaming not reporting any violation

### ISSUE THREAD

**Issue by varunmendiratta** [detekt/detekt#4058]
## Expected Behavior
`BooleanPropertyNaming` was added in 1.18.0. After enabling this in `detekt.yml`, the violations should be reported.

## Observed Behavior
Not even a single violation is reported.

## Steps to Reproduce
I have a multi-module project. I have even setup custom Detekt rules in my project.
Here is the root level build.gradle.
```
plugins {
    id "io.gitlab.arturbosch.detekt" version "1.18.0"
}

apply plugin: 'io.gitlab.arturbosch.detekt'
dependencies {
    detektPlugins project(':static-analysis:detekt-rules') //custom Detekt rules
}

tasks.register("detektAll", Detekt) { //custom Gradle task
    description = "Custom Detekt build for all modules"
    parallel = true
    setSource(file(projectDir))
    config.setFrom(files("$rootDir/detekt.yml"))
    jvmTarget = "1.8"
    classpath.setFrom(project.configurations.getByName("detekt"))
    include("**/*.kt")
    exclude("**/build/**")
    reports {
        html {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.html")
        }
        xml {
            enabled = true
            destination = file("$rootDir/build/reports/detekt-results.xml")
        }
        txt.enabled = false
    }
}
```

Part of `detekt.yml`
```
naming:
  active: true
  severity: error
  BooleanPropertyNaming:
    active: true
  ClassNaming:
    active: true
    classPattern: '[A-Z][a-zA-Z0-9]*'
  ForbiddenClassName:
    active: false
    forbiddenName: [Manager]
  MemberNameEqualsClassName:
    active: true
    ignoreOverridden: true
  NoNameShadowing:
    active: true
  NonBooleanPropertyPrefixedWithIs:
    active: true
```
    
When I copy the code for `BooleanPropertyNaming` in my custom rules project `static-analysis:detekt-rules`, the violations are reported.

## Context
Just trying to enable `BooleanPropertyNaming` on my project.

## Your Environment
* Version of detekt used: 1.18.0
* Version of Gradle used (if applicable): 7.0.0
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS 10.15.6
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
That's a real bug. The fix is in #4071

**Comment by denisstarodubov** [comment]
In 1.19.0 `BooleanPropertyNaming` still doesn't working.

My task: 
```
task detektAll(type: io.gitlab.arturbosch.detekt.Detekt) {
    description = "Runs Detekt task for all modules"
    buildUponDefaultConfig = true
    allRules = true
    parallel = true
    setSource(files(rootDir))
    config.from(files("$rootDir/tools/detekt/detekt.yml"))
    baseline.set(file("$rootDir/tools/detekt/baseline.xml"))
    include("**/*.kt")
    include("**/*.kts")
    exclude("**/resources/**")
    exclude("**/build/**")
    reports {
        html.enabled = true
        xml.enabled = false
        txt.enabled = false
        sarif.enabled = false
    }
}
```

detekt.yml part: 
```
naming:
  active: true
  BooleanPropertyNaming:
    active: true
    excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
    allowedPattern: '^(is|has|are|should|have|can|was|had)'
```

Tried to remove allowPattern block, but nothing changed. 
Copied class `BooleanPropertyNaming` to my custom rule, but nothing changed. While debugging I found out that `bindingContext` is always `BindingContext.EMPTY`

- Version of detekt used: 1.19.0
- Version of Gradle used (if applicable): 7.0.4
- Operating System and version: MacOS 12.2.1

**Comment by cortinico** [comment]
> task detektAll(type: io.gitlab.arturbosch.detekt.Detekt) {

You're not using Type Resolution, therefore you're not running `BooleanPropertyNaming` to it's full capacity:

More on type resolution here https://detekt.dev/type-resolution.html

**Comment by denisstarodubov** [comment]
Thanks, I missed it.
